js/gatecoin.js

```diff
@@ -350,9 +350,9 @@ module.exports = class gatecoin extends Exchange {
             'side': side,
             'price': trade['price'],
             'amount': trade['quantity'],
-            'cost': trade['price'] * trade['quantity'] + trade['feeAmount'],
+            'cost': trade['price'] * trade['quantity'] + (trade['feeAmount'] ? trade['feeAmount'] : 0),
             'fee': {
-                'cost': trade['feeAmount'],
+                'cost': (trade['feeAmount'] ? trade['feeAmount'] : undefined),
```

Comment:
This will still not work in Python, because when you access a non-existent key it will throw a `KeyError` exception, thus you really have to use the `.safe*` family or check `if ('feeAmount' in trade)` otherwise...